### ISSUE TITLE
Prometheus support ignores passed MetricsProvider unless Tendermint runs its own Prometheus server

### ISSUE THREAD

**Issue by silasdavis** [tendermint/tendermint#2019]
Burrow runs its own Prometheus server so I was pleased to see that I could pass a `MetricsProvider`. However in order to get Tendermint to use my provider I must enable:

https://github.com/tendermint/tendermint/blob/master/node/node.go#L238-L242
```go
	if config.Instrumentation.Prometheus {
		csMetrics, p2pMetrics, memplMetrics = metricsProvider()
	} else {
		csMetrics, p2pMetrics, memplMetrics = NopMetricsProvider()
	}
```

But when the node starts that same switches enables Tendermint's own metrics server
https://github.com/tendermint/tendermint/blob/master/node/node.go#L462-L464
```go
	if n.config.Instrumentation.Prometheus {
		n.prometheusSrv = n.startPrometheusServer(n.config.Instrumentation.PrometheusListenAddr)
	}
```

I am already running prometheus and want to run all my metrics through one endpoint. There should either be independent switches or preferably, since you have to pass the node a metrics provider (and you can provide a noop one, just drop the first switch and make `Instrumentation.Prometheus` just enable the Tendermint-hosted server - if you want no metrics to be used just pass the noop.

**Comment by melekes** [comment]
Yeah, I did not think of that. Thank you for raising this.

What if we add a condition to the second code block:

```go
	if n.config.Instrumentation.Prometheus && n.config.Instrumentation.PrometheusListenAddr != "" {
		n.prometheusSrv = n.startPrometheusServer(n.config.Instrumentation.PrometheusListenAddr)
	}
```

That way, if you don't want to run a server, you just `prometheus_listen_addr=""` in your config. How does that sound to you?

**Comment by silasdavis** [comment]
That would work similar to RPC, but why override the metrics provider that is passed in in the code block above before that? Seems like we can indicate noop ourselves by passing that, but whatever we pass should be honoured. In fact we could do the whole thing on checking `n.config.Instrumentation.PrometheusListenAddr != ""` and just using whatever metric provider is passed (which may be noop)...

**Comment by melekes** [comment]
OK, I think I know what to do. We can move if condition into DefaultMetricsProvider() and check for `n.config.Instrumentation.PrometheusListenAddr != ""` before we start the metrics server.

**Comment by melekes** [comment]
Released in v0.22.6.
